### DISCUSSION TITLE
Enhancements Request: Support for Larger Projects and Custom Application Icons

### DISCUSSION THREAD

**Discussion by wladradchenko** [beeware/briefcase#1441] [Ideas]
Hello Briefcase maintainers and contributors,

First and foremost, I want to express my heartfelt gratitude to all of you who have contributed to the Briefcase project. It has been a pivotal tool in the development and distribution of my speech synthesis and cloning open-source project [Wunjo AI](https://github.com/wladradchenko/wunjo.wladradchenko.ru), facilitating the seamless creation of an out-of-the-box installer for the project.

As my application has grown in complexity and size, I have encountered a couple of constraints that I wanted to bring to your attention, in hopes that future versions of Briefcase might address these:

MSI Installer Size Limitation on Windows: Currently, it seems that creating an MSI installer on Windows is not feasible for projects where the total size including all libraries exceeds 2GB. I am keen on learning if there are any plans to enhance Briefcase to support larger project sizes, perhaps by increasing this limit or by introducing the ability to create EXE files as an alternative.

Custom Application Icon: Despite specifying a custom application icon in the pyproject.toml file, the default bee icon persists post-build. I've attempted specifying paths to both individual files and icon sets to no avail. It would be greatly appreciated if you could guide me on how to correctly set up a custom icon, or consider this as a feature request for future releases.

I am eager to hear your thoughts on these matters

Thank you once again for your hard work and dedication to this project.

**Comment by freakboy3742** [comment]
The size limitation on Windows MSIs is a known issue - see #646 for details. The MSI format has limitations on the number of files, and the overall size of the underlying archive. Wix4 (#1185) *might* have a workaround for this, but initial exploration into adopting Wix4 haven't been very successful. We'd like to fix this, but Wix is not well documented, so we haven't had a lot of luck working out what is required.

We don't have an EXE-based installer - although I'd suggest that if a MSI-based installer isn't working because of file size, an EXE-based installer is even *less* likely to work. 

We do have a ZIP installer - that isn't really an "installer" per se, but it does allow you to distribute a project as a single file.

As for custom icons - you'll need to provide details. We've got plenty of evidence that custom icons can be used; every report we've had of icons "not working" has been the result of a either:
1. A configuration error;
2. A failure to run `briefcase update --update-resources`;
3. A failure to provide the right files (and/or ignoring the error messages provided as part of (2)).

If you can tell us what you've done, what files you've put where, and provide the log file from a fresh `briefcase create --log` on your project, we can help diagnose what is going on.

**Reply by wladradchenko** [reply]
In pyproject.toml I set `icon = "src/wunjo/static/basic/icon/gray"`. In src/wunjo/static/basic/icon/gray I have those files on screen:

![Screenshot from 2023-09-11 11-08-22](https://github.com/beeware/briefcase/assets/56233697/adc0f0c8-b121-4d0a-b581-c88861f33844)

I have tried to set for Windows, Ubuntu and MacOS path to this folder or path to file for specific icon, for example for MacOS `icon = "src/wunjo/static/basic/icon/gray/logo.icns"`. I have tried also to do folder on Ubuntu similar folder "hicolor" as on screen:

![Screenshot from 2023-09-11 11-10-48](https://github.com/beeware/briefcase/assets/56233697/d02d3889-3d27-4677-9a19-f9707df4e95d)

Anyway, the result has been one, I get default icon of beeware.

**Reply by freakboy3742** [reply]
When we ask for the log, we really need the *whole* log, not just the parts you think are important.

In particular, I suspect your log will contain content that looks something like:
```
[helloworld] Installing application resources...
Unable to find src/wunjo/static/basic/icon/gray-16.png for 16px application icon; using default
Unable to find src/wunjo/static/basic/icon/gray-32.png for 32px application icon; using default
Unable to find src/wunjo/static/basic/icon/gray-64.png for 64px application icon; using default
Unable to find src/wunjo/static/basic/icon/gray-128.png for 128px application icon; using default
Unable to find src/wunjo/static/basic/icon/gray-256.png for 256px application icon; using default
Unable to find src/wunjo/static/basic/icon/gray-512.png for 512px application icon; using default
```

which will tell you what icon files it was looking for, and what files it failed to find.

**Reply by wladradchenko** [reply]
Thank you. After this answer I understand what it a mistake was. I changed on `icon = "src/wunjo/static/basic/icon/gray/logo"` and set in folder those files on screen. Now it works fine.

![Screenshot from 2023-09-11 20-57-17](https://github.com/beeware/briefcase/assets/56233697/395e2b3c-0382-44e5-9284-07fe2798e092)
